motionplan/motionPlanner.go

```diff
@@ -427,36 +427,45 @@ func CheckPlan(
 	// The solver frame will have had its PTGs filled in the newPlanManager() call, if applicable.
 	relative := len(sf.PTGSolvers()) > 0
 
+	// where ought the robot be on the plan
+	pathPosition := spatialmath.PoseBetweenInverse(errorState, currentPosition)
+
 	if relative {
 		// get pose of robot along the current trajectory it is executing
 		lastPose, err := sf.Transform(currentInputs)
 		if err != nil {
 			return err
 		}
 
-		// where ought the robot be on the plan
-		pathPosition := spatialmath.PoseBetweenInverse(errorState, currentPosition)
-
 		// absolute pose of the previous node we've passed
 		formerRunningPose := spatialmath.PoseBetweenInverse(lastPose, pathPosition)
 
-		// convert planNode's poses to be in absolute corrdinated
+		// convert planNode's poses to be in absolute coordinates
 		if planNodes, err = rectifyTPspacePath(planNodes, sf, formerRunningPose); err != nil {
 			return err
 		}
 	}
+
 	// adjust planNodes by the errorState
+	// this only changes a node's pose and not its inputs
 	planNodes = transformNodes(planNodes, errorState)
 
 	// pre-pend node with current position of robot to planNodes
 	// Note that currentPosition is assumed to have already accounted for the errorState
-	planNodes = append([]node{&basicNode{pose: currentPosition, q: currentInputs}}, planNodes...)
+	if relative {
+		planNodes = append([]node{&basicNode{pose: currentPosition, q: currentInputs}}, planNodes...)
```

Comment:
Assuming `currentInputs` is the result of calling `base.CurrentInputs()`, this isn't correct and will yield the wrong thing downstream. Will have to come back to this once I think more about what the right thing is.

The reason this is probably incorrect, is the current inputs indicate what the base has already executed, not what it is currently/about to execute.